pkg/internal/goexec/structmembers.go

```diff
@@ -443,7 +443,7 @@ func structMemberOffsetsFromDwarf(data *dwarf.Data) (FieldOffsets, map[GoOffset]
 		log.Debug("inspecting fields for struct type", "type", typeName)
 		if err := readMembers(reader, structMember.fields, expectedReturns, fieldOffsets); err != nil {
 			log.Debug("error reading DWARF info", "type", typeName, "error", err)
-			return nil, expectedReturns
+			return fieldOffsets, expectedReturns
```

Comment:
I believe there's more to it. This lies in the case when `instrumentable.Offsets` is `null`. The stack posted on the issue points that [this call](https://github.com/grafana/beyla/blob/119226432feff14ed60bd404f615d0ee84e3c671/pkg/internal/ebpf/tracer_linux.go#L266) is passing a nill `Offsets` pointer. This pointer gets initialised [here](https://github.com/grafana/beyla/blob/119226432feff14ed60bd404f615d0ee84e3c671/pkg/internal/discover/typer.go#L119). Here we are initialising `goexec.Offsets.FieldOfssets` which is not directly involving in the crash (it's the toplevel `goexec.Offsets`  being nil that is causing the crash [here](https://github.com/grafana/beyla/blob/119226432feff14ed60bd404f615d0ee84e3c671/pkg/internal/ebpf/gotracer/gotracer.go#L156) .

We opted to leave this in the PR because this fixes the issue you are talking about, but I reckon that happens to be a different issue.